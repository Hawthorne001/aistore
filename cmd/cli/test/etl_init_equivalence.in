// RUN k8s
# Test that 'ais etl init' is equivalent to 'ais etl init spec'
curl https://raw.githubusercontent.com/NVIDIA/ais-etl/main/transformers/hello_world/pod.yaml -o hello_world_spec.yaml -s // IGNORE

# Create ETL using 'ais etl init spec' (explicit)
ais etl init spec --spec="hello_world_spec.yaml" --comm-type="hpush://" --name=$ETL_NAME-explicit // IGNORE

# Create ETL using 'ais etl init' (implicit spec, new functionality)
ais etl init --spec="hello_world_spec.yaml" --comm-type="hpush://" --name=$ETL_NAME-implicit // IGNORE

# Verify both ETLs are running
ais etl show | grep $ETL_NAME-explicit | awk '{print $2}'
ais etl show | grep $ETL_NAME-implicit | awk '{print $2}'

# Show specification for both ETLs to verify they have identical properties
ais etl show $ETL_NAME-explicit | grep -m1 "communication_type"
ais etl show $ETL_NAME-implicit | grep -m1 "communication_type"

# Clean up
ais etl stop $ETL_NAME-explicit // IGNORE
ais etl rm $ETL_NAME-explicit // IGNORE
ais etl stop $ETL_NAME-implicit // IGNORE
ais etl rm $ETL_NAME-implicit // IGNORE
rm -f hello_world_spec.yaml // IGNORE
