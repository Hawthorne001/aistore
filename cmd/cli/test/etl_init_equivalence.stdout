Running
Running
communication_type: hpush://
communication_type: hpush://